fea-rs/src/compile/tables/base.rs

```diff
@@ -17,24 +53,73 @@ pub(crate) struct ScriptRecord {
     pub values: Vec<i16>,
 }
 
+#[derive(Clone, Debug)]
+pub(crate) struct MinMax {
+    pub min: i16,
+    pub max: i16,
+    // we don't implement per-feature minmax, matching fonttools
+    //pub _features: Vec<FeatMinMax>,
+}
+
+#[derive(Clone, Debug)]
+#[expect(dead_code, reason = "not yet implemented")]
+pub(crate) struct FeatMinMax {
+    pub feature: Tag,
+    pub min: i16,
+    pub max: i16,
+}
+
 impl BaseBuilder {
     pub(crate) fn build(&self) -> write_base::Base {
         let mut result = write_base::Base::default();
         if !self.horiz_tag_list.is_empty() {
             assert!(!self.horiz_script_list.is_empty(), "validate this");
-            let haxis = BaseBuilder::build_axis(&self.horiz_tag_list, &self.horiz_script_list);
+            let haxis = BaseBuilder::build_axis(
+                &self.horiz_tag_list,
+                &self.horiz_script_list,
+                &self.horiz_min_max,
+            );
             result.horiz_axis = haxis.into();
         }
         if !self.vert_tag_list.is_empty() && !self.vert_script_list.is_empty() {
             assert!(!self.vert_script_list.is_empty(), "validate this");
-            let vaxis = BaseBuilder::build_axis(&self.vert_tag_list, &self.vert_script_list);
+            let vaxis = BaseBuilder::build_axis(
+                &self.vert_tag_list,
+                &self.vert_script_list,
+                &self.vert_min_max,
+            );
             result.vert_axis = vaxis.into();
         }
         result
     }
 
-    fn build_axis(tag_list: &[Tag], script_list: &[ScriptRecord]) -> write_base::Axis {
+    fn build_axis(
+        tag_list: &[Tag],
+        script_list: &[ScriptRecord],
+        minmax: &HashMap<Tag, Vec<(Tag, MinMax)>>,
+    ) -> write_base::Axis {
+        // a little helper fn
+        fn get_dflt_and_lang_sys_minmax(
+            minmax: Option<&Vec<(Tag, MinMax)>>,
+        ) -> (Option<write_base::MinMax>, Vec<BaseLangSysRecord>) {
+            let mut dflt = None;
+            let mut rest = Vec::new();
+
+            for (tag, table) in minmax.into_iter().flatten() {
+                if *tag == Tag::new(b"dflt") {
+                    dflt = Some(table.compile());
+                } else {
+                    rest.push(BaseLangSysRecord::new(*tag, table.compile()));
+                }
+            }
+
+            (dflt, rest)
```

Comment:
BaseLangSysRecords should be sorted "in alphabetical order by BaseLangSysTag"

https://learn.microsoft.com/en-us/typography/opentype/spec/base

I think we aren't sorting them here (and write-fonts isn't either?) so if fonts have multiple language specific minmax entries for the same scripts we'd keep them in insertion order rather than alphabetical